tests/lib/azure/vnet.go

```diff
@@ -23,15 +24,62 @@ func NewVnet(f fixture.F) *Vnet {
 	return as
 }
 
+func validateVnetDnsServers(
+	t *testing.T,
+	expectedDnsServers []string,
+	net network.VirtualNetwork,
+) error {
+	if net.DhcpOptions == nil {
+		return errors.New("The field DhcpOptions is nil!")
+	}
+	if net.DhcpOptions.DNSServers == nil {
+		return errors.New("The field DNSServers is nil!")
+	}
+
+	dnsServers := *net.DhcpOptions.DNSServers
+	err := fmt.Errorf(
+		"expected DNS servers[%s], got [%s]",
+		expectedDnsServers,
+		dnsServers,
+	)
+	if len(dnsServers) != len(expectedDnsServers) {
+		return err
+	}
+	for _, expectedDnsServer := range expectedDnsServers {
+		found := false
+		for _, dnsServer := range dnsServers {
```

Comment:
uaheuaheuaheuaheuahue

I was going to say that I implemented it to support different order on purpose. Sometimes I miss the concept of a set with equality defined in Go, this is the perfect case for it, I'm not interested on order just on the set of values being the same.

Even using maps to implement sets won't cut it since maps are not comparable =/